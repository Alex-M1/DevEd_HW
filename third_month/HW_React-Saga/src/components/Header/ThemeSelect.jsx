import React from 'react';
import { useTheme } from '../../helpers/withTheme';

function ThemeSelect() {
  const { theme, changeTheme } = useTheme();
  return (
    <select value={theme} onChange={changeTheme}>
      <option value='dark'>dark</option>
      <option value='light'>light</option>
    </select>
  );
}

export default ThemeSelect;
