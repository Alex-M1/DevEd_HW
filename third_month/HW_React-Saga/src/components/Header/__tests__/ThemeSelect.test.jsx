import React from 'react';
import ThemeSelect from '../ThemeSelect';

const theme = 'dark';
jest.mock('../../../helpers/withTheme', () => ({
  useTheme: jest.fn().mockReturnValue({
    theme,
    changeTheme: jest.fn().mockReturnValue(theme),
  }),
}));
describe('ThemeSelect', () => {
  it('Should match snapshot', () => {
    const component = global.shallowSmart(<ThemeSelect />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have select', () => {
    const component = global.mountSmart(<ThemeSelect />);
    expect(component.find('select')).toHaveLength(1);
  });
  it('Should have option', () => {
    const component = global.mountSmart(<ThemeSelect />);
    expect(component.find('option')).toHaveLength(2);
  });
  it('Should have theme', () => {
    const component = global.mountSmart(<ThemeSelect />);
    component.find('select').simulate('change', { target: { value: theme } });
    expect(component.find('select').getDOMNode().value).toBe(theme);
  });
});
