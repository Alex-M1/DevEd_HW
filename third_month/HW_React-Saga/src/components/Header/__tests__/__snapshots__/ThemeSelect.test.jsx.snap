// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSelect Should match snapshot 1`] = `"<select><option selected=\\"\\" value=\\"dark\\">dark</option><option value=\\"light\\">light</option></select>"`;
